wagtail/wagtailcore/templatetags/wagtailcore_tags.py

```diff
@@ -46,3 +48,63 @@ def richtext(value):
         html = expand_db_html(value)
 
     return mark_safe('<div class="rich-text">' + html + '</div>')
+
+
+class IncludeBlockNode(template.Node):
+    def __init__(self, block_var, extra_context, use_parent_context):
+        self.block_var = block_var
+        self.extra_context = extra_context
+        self.use_parent_context = use_parent_context
+
+    def render(self, context):
+        try:
+            value = self.block_var.resolve(context)
+        except template.VariableDoesNotExist:
+            return ''
+
+        if hasattr(value, 'render_as_block'):
+            if self.use_parent_context:
+                new_context = context.flatten()
+            else:
+                new_context = {}
+
+            if self.extra_context:
+                for var_name, var_value in self.extra_context.items():
+                    new_context[var_name] = var_value.resolve(context)
+
+            return value.render_as_block(context=new_context)
+        else:
+            return force_text(value)
+
+
+@register.tag
+def include_block(parser, token):
```

Comment:
It seems to me that we need to add a similar tag for the Jinja2 template engine.